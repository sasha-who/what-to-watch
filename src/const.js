export const RECOMENDED_FILMS_COUNT = 4;
export const PREVIEW_DELAY = 1000;
export const REVIEWS_IN_COLUMN_COUNT = 3;
export const DEFAULT_GENRE = `all genres`;
export const MAX_FILTERS_COUNT = 10;
export const INITIAL_FILMS_COUNT = 8;
export const ADDITIONAL_FILMS_COUNT = 8;
export const DURATION_INITS = `minutes`;
export const RUN_TIME_FORMAT = `HH:mm:ss`;
export const VIDEO_CLASS_NAME = `player__video`;
export const SECONDS_IN_MINUTE_COUNT = 60;
export const SERVER_TIMEOUT = 5000;
export const BASE_URL = `https://4.react.pages.academy/wtw`;
export const RATING_START_COUNT = 5;
export const STAR_CHECKED_BY_DEFAULT = 3;
export const REVIEW_MIN_LENGTH = 50;

export const RatingRange = {
  MIN: 0,
  MAX_BAD: 3,
  MAX_NORMAL: 5,
  MAX_GOOD: 8,
  MAX: 10
};

export const RatingGrade = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`
};

export const Screen = {
  MAIN: `main`,
  CARD: `card`,
  SIGN_IN: `sign-in`,
  REVIEW: `review`
};

export const TabsNames = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const TabsData = [
  {name: TabsNames.OVERVIEW, content: `Overview`},
  {name: TabsNames.DETAILS, content: `Details`},
  {name: TabsNames.REVIEWS, content: `Reviews`}
];

export const PlayIconStart = {
  WIDTH: 19,
  HEIGHT: 19,
  ID: `play-s`,
  DESCRIPTION: `Play`
};

export const PlayIconPause = {
  WIDTH: 14,
  HEIGHT: 21,
  ID: `pause`,
  DESCRIPTION: `Pause`
};

export const ReviewDate = {
  HUMAN_FORMAT: `MMMM D, YYYY`,
  SERVICE_FORMAT: `YYYY-MM-DD`
};

export const AuthorizationStatus = {
  AUTHORIZED: `AUTHORIZED`,
  NO_AUTHORIZED: `NO_AUTHORIZED`,
};

export const LoaderData = {
  TYPE: `Circles`,
  COLOR: `#dfcf77`,
  WIDTH: 150,
  HEIGHT: 150,
  STYLE: {
    position: `fixed`,
    top: `50%`,
    left: `50%`,
    transform: `translate(-50%, -50%)`
  }
};

export const HttpStatus = {
  SUCCESS: 200,
  REDIRECT: 300,
  UNAUTHORIZED: 401
};

export const CommentPostStatus = {
  PENDING: `PENDING`,
  POSTING: `POSTING`,
  OK: `OK`,
  ERROR: `ERROR`
};
