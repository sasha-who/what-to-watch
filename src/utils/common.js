import moment from "moment";
import {
  DURATION_INITS,
  RUN_TIME_FORMAT,
  DEFAULT_GENRE,
  RECOMENDED_FILMS_COUNT
} from "../const.js";

export const getRandomIntegerNumber = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);

  return Math.floor(Math.random() * (max - min + 1)) + min;
};

export const getRandomArrayItem = (elements) => {
  const randomIndex = getRandomIntegerNumber(0, elements.length - 1);

  return elements[randomIndex];
};

const shuffleArray = (array) => {
  for (let i = array.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));

    [array[i], array[j]] = [array[j], array[i]];
  }

  return array;
};

export const getRandomArrayItems = (array) => {
  const shuffledArray = shuffleArray(array.slice());

  return shuffledArray.slice(0, getRandomIntegerNumber(0, array.length));
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getFormatedRunTime = (duration) => {
  const durationInMinutes = moment.duration(duration, DURATION_INITS);

  return `${durationInMinutes.hours()}h ${durationInMinutes.minutes()}m`;
};

export const getRunTimeForPlayer = (duration) => {
  const durationInMillisecond = moment.duration(duration, DURATION_INITS).asMilliseconds();

  return moment.utc(durationInMillisecond).format(RUN_TIME_FORMAT);
};

export const getFilmsFilteredByGenre = (allFilms, genre) => {
  if (genre === DEFAULT_GENRE) {
    return allFilms;
  }

  return allFilms.filter((film) => film.genre.toLowerCase() === genre);
};

export const getSimilarForCurrentFilm = (allFilms, currentFilm) => {
  return (
    allFilms.filter((film) => (film !== currentFilm) && (film.genre === currentFilm.genre))
      .slice(0, RECOMENDED_FILMS_COUNT)
  );
};

export const getFilmFromParameters = (films, id) => {
  const index = films.findIndex((film) => film.id === Number(id));

  return films[index];
};
