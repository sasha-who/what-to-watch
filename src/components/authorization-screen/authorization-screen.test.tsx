import * as React from "react";
import * as renderer from "react-test-renderer";
import {Router} from "react-router-dom";
import history from "../../history";
import {AuthorizationStatus, HttpStatus} from "../../const";
import AuthorizationScreen from "../authorization-screen/authorization-screen";
import {noop} from "../../utils/common";

it(`AuthorizationScreen without errors should render correctly`, () => {
  const tree = renderer.create(
      <Router
        history={history}
      >
        <AuthorizationScreen
          isInputValid={true}
          authorizationStatus={AuthorizationStatus.AUTHORIZED}
          onAuthorizationFormSubmit={noop}
          onInputValidityChange={noop}
        />
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`AuthorizationScreen with validation error should render correctly`, () => {
  const tree = renderer.create(
      <Router
        history={history}
      >
        <AuthorizationScreen
          isInputValid={false}
          authorizationStatus={AuthorizationStatus.AUTHORIZED}
          onAuthorizationFormSubmit={noop}
          onInputValidityChange={noop}
        />
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`AuthorizationScreen with server error should render correctly`, () => {
  const tree = renderer.create(
      <Router
        history={history}
      >
        <AuthorizationScreen
          isInputValid={false}
          authorizationStatus={AuthorizationStatus.AUTHORIZED}
          onAuthorizationFormSubmit={noop}
          onInputValidityChange={noop}
          loginError={HttpStatus.NOT_FOUND}
        />
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
