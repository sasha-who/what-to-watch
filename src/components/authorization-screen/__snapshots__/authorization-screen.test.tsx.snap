// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorizationScreen with server error should render correctly 1`] = `null`;

exports[`AuthorizationScreen with validation error should render correctly 1`] = `null`;

exports[`AuthorizationScreen without errors should render correctly 1`] = `null`;
